src/api/donations.py

```
@@ -0,0 +1,618 @@
+            await _on_payment_failed(env, obj)
+        elif event_type == "customer.subscription.created":
+            await _on_subscription_created(env, obj)
+        elif event_type == "customer.subscription.updated":
+            await _on_subscription_updated(env, obj)
+        elif event_type == "customer.subscription.deleted":
+            await _on_subscription_cancelled(env, obj)
+        elif event_type == "invoice.payment_succeeded":
+            await _on_invoice_paid(env, obj)
+        elif event_type == "invoice.payment_failed":
+            await _on_invoice_failed(env, obj)
+    except Exception as exc:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)